### ISSUE TITLE
Commodity Forwards Notional Resets (bug?)

### ISSUE THREAD

**Issue by Nick-Git2** [OpenSourceRisk/Engine#203]
Hi there,

I've set up a new commodity forward and run standard stuff similar to Example_24. I've noticed that the notional gets set to the present forward price x quantity similar as if it was resettable daily  (see for example notional from NPV of Example_24).

In my opinion, the default for a forward notional should be the forward's agreed price at closure ("strike") x quantity and this notional should not change during the trade's lifetime.

Here's a modified Example_24 (trade's price/"strike" set to 1100) to highlight the issue. Forward from Curve is at 1161.80:
![grafik](https://github.com/OpenSourceRisk/Engine/assets/65270786/9fdc814b-2d13-49b4-87cd-338f54649e6c)

Rgds,

Nick

**Comment by rkapl123** [comment]
Can you post your portfolio.xml?

**Comment by Nick-Git2** [comment]
> Can you post your portfolio.xml?

It's pretty much an excerpt from portfolio.xml from Example_24. Only "strike" and quantity changed. All other input parameters/files unchanged.

<Portfolio>
  <Trade id="CommodityForward_Gold_Apr_17">
    <TradeType>CommodityForward</TradeType>
    <Envelope>
      <CounterParty>CPTY_A</CounterParty>
      <NettingSetId>CPTY_A</NettingSetId>
      <AdditionalFields/>
    </Envelope>
    <CommodityForwardData>
      <Position>Long</Position>
      <Maturity>2017-04-28</Maturity>
      <Name>COMDTY_GOLD_USD</Name>
      <Currency>USD</Currency>
      <Strike>1100</Strike>
	  <Quantity>1000</Quantity>
    </CommodityForwardData>
  </Trade>
</Portfolio>

**Comment by rkapl123** [comment]
You're right, the problem seems to be in the implementation of CommodityForward::notional()
```
Real CommodityForward::notional() const {
    // try to get the notional from the additional results of the instrument
    try {
        return instrument_->qlInstrument(true)->result<Real>("currentNotional");
    } catch (const std::exception& e) {
        if (strcmp(e.what(), "currentNotional not provided"))
            ALOG("error when retrieving notional: " << e.what());
    }
    // if not provided, return null
    return Null<Real>();
}
```

**Comment by pcaspers** [comment]
I agree, the notional should most likely be "quantity x strike" as here

https://github.com/OpenSourceRisk/Engine/blob/master/OREData/ored/portfolio/commodityforward.cpp#L149
 
In general we set the notional in a way that complies with ISDA recommendations for trade reconciliation. We'll confirm that "quantity x strike" is correct and fix this in the next release. Of course you are welcome to open a PR if you want.
